File: app/lib/core/theme/sbc_colors.dart
import 'package:flutter/material.dart';

/// SBC (Sniper Business Center) brand palette, taken from the production
/// `sbcprecom` and `sbc-live` web projects. Seed for the M3 theme.
class SbcColors {
  const SbcColors._();

  // Primary — blue
  static const Color primary = Color(0xFF1862F0);
  static const Color primaryLight = Color(0xFF4285F4);
  static const Color primaryDark = Color(0xFF0F4FB3);

  // Secondary — green
  static const Color secondary = Color(0xFF92B127);
  static const Color secondaryLight = Color(0xFFA5C142);
  static const Color secondaryDark = Color(0xFF7A951F);

  // Accent — orange
  static const Color accent = Color(0xFFF49101);
  static const Color accentLight = Color(0xFFFFA726);
  static const Color accentDark = Color(0xFFE57C00);

  // Semantic
  static const Color success = Color(0xFF10B981);
  static const Color warning = Color(0xFFF59E0B);
  static const Color error = Color(0xFFEF4444);
  static const Color info = Color(0xFF3B82F6);
  static const Color live = Color(0xFFE5484D);

  /// WhatsApp brand green — used only for the "contacter sur WhatsApp"
  /// affordance, so the action is recognisable at a glance (cahier §8).
  static const Color whatsapp = Color(0xFF25D366);

  // Surfaces
  static const Color bgLight = Color(0xFFF4F7FD);
  static const Color surfaceLight = Color(0xFFFFFFFF);
  static const Color surfaceTint = Color(0xFFEDF2FB);
  static const Color bgDark = Color(0xFF0B1120);
  static const Color surfaceDark = Color(0xFF121A2E);

  /// Signature 3-stop brand arc: blue → green → orange.
  static const LinearGradient brandArc = LinearGradient(
    colors: [primary, secondary, accent],
    stops: [0, 0.55, 1],
  );
}

File: ios/SBCContacts/Core/Theme/SBCColors.swift
import SwiftUI

/// SBC (Sniper Business Center) brand palette, taken from the production
/// `sbcprecom` and `sbc-live` web projects.
enum SBCColors {
    // Primary — blue
    static let primary = Color(hex: 0x1862F0)
    static let primaryLight = Color(hex: 0x4285F4)
    static let primaryDark = Color(hex: 0x0F4FB3)

    // Secondary — green
    static let secondary = Color(hex: 0x92B127)
    static let secondaryLight = Color(hex: 0xA5C142)
    static let secondaryDark = Color(hex: 0x7A951F)

    // Accent — orange (Material "tertiary")
    static let accent = Color(hex: 0xF49101)
    static let accentLight = Color(hex: 0xFFA726)
    static let accentDark = Color(hex: 0xE57C00)

    // Semantic
    static let success = Color(hex: 0x10B981)
    static let warning = Color(hex: 0xF59E0B)
    static let error = Color(hex: 0xEF4444)
    static let info = Color(hex: 0x3B82F6)
    static let live = Color(hex: 0xE5484D)

    /// WhatsApp brand green — used only for the "contacter sur WhatsApp"
    /// affordance, so the action is recognisable at a glance (cahier §8).
    static let whatsapp = Color(hex: 0x25D366)

    // Surfaces
    /// System grouped surfaces, so every screen sits on the same native ground
    /// as the grouped lists, sheets and search bar.
    static let background = Color(.systemGroupedBackground)
    static let surface = Color(.secondarySystemGroupedBackground)
    static let surfaceTint = Color(hex: 0xEDF2FB)

    // Material-role equivalents used by the Flutter theme
    static let onSurface = Color(hex: 0x1A1C22)
    static let onSurfaceVariant = Color(hex: 0x5B6170)
    static let outlineVariant = Color(hex: 0xC6CBD6)
    static let primaryContainer = Color(hex: 0xD9E4FD)
    static let onPrimaryContainer = Color(hex: 0x0A2E73)
    static let secondaryContainer = Color(hex: 0xE7F0C8)
    static let onSecondaryContainer = Color(hex: 0x2F3A08)

    /// Signature 3-stop brand arc: blue → green → orange.
    static let brandArc = LinearGradient(
        stops: [
            .init(color: primary, location: 0),
            .init(color: secondary, location: 0.55),
            .init(color: accent, location: 1),
        ],
        startPoint: .leading,
        endPoint: .trailing
    )
}

extension Color {
    init(hex: UInt32, opacity: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: opacity
        )
    }
}
